### PR TITLE
Change short name to vc-jose-cose

### PR DESCRIPTION
**Pull request by mprorock**
putting this adjustment in its own pr

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-jwt/pull/115.html" title="Last updated on Jun 30, 2023, 12:24 PM UTC (500588f)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-jwt/115/289cd68...500588f.html" title="Last updated on Jun 30, 2023, 12:24 PM UTC (500588f)">Diff</a>

### CODE REVIEW

**Review by OR13 — APPROVED** [review]

File: index.html
Comment (TallTed) [inline index.html:22]: This seems to suggest that the <https://github.com/w3c/vc-jwt> repo should be moved to <https://github.com/w3c/vc-jose-cose>?

Comment (OR13) [reply index.html:22]: yep, we would want to do that, after the working group approves.

Comment (iherman) [reply index.html:22]: Well... as long as this is only a change of the Repo, that is a piece of cake.

However, does it mean to change the short name of the specification as well? We _can_ do that, but it requires all kinds of complicated set ups at the W3C system team level, because they will have to set up redirections. Remember that no W3C URL should be a 404, ie, the `https://www.w3.org/TR/vc-jwt` should live forever in the form of a redirection. In essence, we would have two /TR URLs for the same document, even if the `vc-jwt` version would not be advertised.

Is this really worth the head ache and the trouble?

Comment (andresuribe87) [reply index.html:22]: I think renaming this is worth it. 

As far as _how_, perhaps it would be easier to create a new spec. Then in the text of the vc-jwt spec point to the new one, without redirecting?

Comment (OR13) [reply index.html:22]: @iherman imo it is worth it for a lot of reasons, many of which have been documented in the most contentious PRs that we are not able to merge.

The current name no longer reflects working group consensus, based on resolutions we have passed, the discussions regarding mappings, and advancements that have been made outside the working group.

I leave it to the chairs to confirm, or clarify what I am summarizing.

@Sakurann @brentzundel

Comment (brentzundel) [reply index.html:22]: It is my understanding that there is general agreement to re-name the VC-JWT spec to be more accurate. Done properly, this would require also changing the short name.
I see two options:
1. withdraw/retire the VC-JWT spec along with the short name `vc-jwt` and add text that explains where the new spec is.
2. rename the repository, introduce a new short name, and add a redirect to `vc-jwt`

from my perspective option 2 is better, but I am not the one who will need to deal with actually making any of this happen.

**Review by selfissued — APPROVED** [review]
I believe this more closely aligns the naming with the substance of the spec.

File: index.html
Comment (mprorock) [reply index.html:22]: I am good with option 2

Comment (Sakurann) [reply index.html:22]: concurring with Brent's assessment.

Comment (Sakurann) [inline index.html:22]: could you elaborate a little more why not `vc-jose`, but `vc-jose-cose`? what is the plan for cose?

Comment (iherman) [reply index.html:22]: I propose that we (which may be me...) contact the webmaster to see what possible hurdles there are on option (2). If they are too important, we may have to fall back on option (1).

Do we have a consensus about trying (2)?

**Review by brentzundel — APPROVED** [review]

**Comment by iherman** [comment]
FYI: I have asked the webmaster to tell me shat it entails to change the short name of a Rec-track publication.

**Comment by iherman** [comment]
Here are the steps that we will have to follow to officially change the short name of the spec as far as the W3C publication system goes:

 1. Switch echidna off
 2. The editor prepares a publishing ready version of the spec using the respec facilities and stores it somewhere where I can access it
 3. I (or the editor) has to modify the version generated by respec by adding a special `data-previous-shortname` attribute at a specific place, see [1] for the details
 4. The document gets published in the old-skool way by pushing it to the W3C server (something I have to do) and ask the webmaster to make an official publication
 5. Once the official publication is done, our echidna script should be modified (see [2]) and then re-instated

Once this process is started, no new PR should be merged until step (5) has been reached, to avoid versioning problems.

This has to be done only once to add the information to the publication change.

[1] https://www.w3.org/pubrules/doc/rules/?profile=WD#docIDHistory
[2] https://github.com/w3c/vc-jwt/blob/197b357151d92f3ec4ca9ef65e69990f2df259db/.github/workflows/auto-publish.yml#L19


cc: @deniak

**Comment by iherman** [comment]
Adding to the practical aspect in https://github.com/w3c/vc-jwt/pull/115#issuecomment-1620408736, I believe that any such change would require a formal resolution of the WG; just as the decision on the choice of the short name, I guess any change thereof should be on record.

cc @Sakurann @brentzundel

File: index.html
Comment (OR13) [reply index.html:22]: We define how to secure the media types registered in https://w3c.github.io/vc-data-model/

We already explain how to secure them as COSE Sign1.

https://w3c.github.io/vc-jwt/#with-cose

We don't want key discovery or other header processing details to be different for JOSE and COSE... we want consistent alignment between protected header values and key and algorithms.

for example:

In jose, protected header contains `alg` and `kid`... `kid` used to discover `jwk` with same `alg`.... 
In cose, protected header contains `alg` and `kid`... `kid` used to discover `cose key` with same `alg`.... 

When `iss` is present in the header, is `kid` relative... see: https://identity.foundation/jwt-vc-presentation-profile/#kid-jose-header

We need to address these usability issues related to W3C specs, consistently for JOSE and COSE.

Comment (OR13) [reply index.html:22]: @Sakurann @brentzundel please establish consensus on this topic.

Comment (brentzundel) [reply index.html:22]: I'll put in on the agenda for next week's VCWG call

**Comment by OR13** [comment]
I will open a PR that does 1-3 here: https://github.com/w3c/vc-jwt/pull/115#issuecomment-1620408736 

We will old all the PRs after that PR is merged, 

We will complete

4 and 5

We will unblock the open PRs.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-07-12-vcwg#section3) on 2023-07-12

List of resolutions:

- ***[Resolution No. 1](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-07-12-vcwg#resolution1):  change the shortname and repository of VC-JWT to vc-jose-cose.***

<details><summary><i>View the transcript</i></summary>

### 3. VC-JWT shortname change to vc-jose-cose.

**Brent Zundel:** there has been discussion of the vc-jwt spec... people seems to like vc-jose-cose.  
… seems to adjust the name to better refect the work that has been done.  

_See github pull request [vc-jwt#115](https://github.com/w3c/vc-jwt/pull/115)._




**Brent Zundel:** seems pretty straightforward.  
… any changes requested?  

**Manu Sporny:** Regarding our charter...  
… our charter doesn't explicitly list COSE as being in scope.  
… However, we note, under the Securing Verifiable Credentials (SVC) 1.0 section of the charter, that "The following are a non-exhaustive selection of expected input documents".  
… and the group has decided that defining JOSE and COSE would be useful.  
… Therefore, the charter allows for this change to be made.  

**Brent Zundel:** yes, our charter says we are allowed to have various input documents.  
… seems cose is in scope, as a possible securing mechanism.  
… for the record, our charter lists JWT in scope, it also says something about PGP.  

> **Proposed resolution: change the shortname and repository of VC-JWT to vc-jose-cose.** *(Brent Zundel)*

> *Brent Zundel:* +1.

> *Orie Steele:* +1.

> *Andres Uribe:* +1.

> *Chris Abernethy:* +1.

> *Ivan Herman:* +1.

> *Manu Sporny:* +0.75 (due to concerns around scope creep).

> *Dave Longley:* +1.

> *Ted Thibodeau Jr.:* +1.

> *Will Abramson:* +1.

> *Gabe Cohen:* +0 it's hard to say.

> *pauld gs1:* +1.

> ***Resolution #1: change the shortname and repository of VC-JWT to vc-jose-cose.***

**Ivan Herman:** yeah so... on the practicalities. I had discussed this with the web master.  
… there is an issue comment: [https://github.com/w3c/vc-jwt/pull/115#issuecomment-1620408736](https://github.com/w3c/vc-jwt/pull/115#issuecomment-1620408736).  
… it describes what has to be done... in terms of changing the repo name / TR... etc... its a bit complicated.  
… who will do the work?  
… I am not sure who is really driving this document.  
… I don't know who my partner will be.  

**Orie Steele:** I'm happy to make the necessary changes, Ivan. Let me know what changes need to be made.  

**Ivan Herman:** thanks orie, please go through the list on the issue comment, sometime next week.  

</details>

**Comment by OR13** [comment]
@iherman instructions linked seem outdated...

>  The syntax of a “history” URI must be https://www.w3.org/standards/history/shortname/, and consistent with the shortname mentioned in 'Latest Version'. Note: If there's a shortname change it must be specified using the following data attribute data-previous-shortname='previous-shortname' on the <a> element.

But I only see

> <a href="https://www.w3.org/standards/history/vc-jwt/" ...

**Comment by OR13** [comment]
@iherman 1-3 are complete over here:

https://github.com/w3c/vc-jwt/pull/125/commits/c6b17eef668bf195ac37d670d52355ed7a606121

Your instructions seem to be lacking a comment on applying the new short name... should be doing that in the same PR?

should I change `vc-jwt` to `vc-jose-cose` in https://github.com/w3c/vc-jwt/pull/125 ?

**Comment by OR13** [comment]
Closing this in favor of https://github.com/w3c/vc-jwt/pull/126

**Pull request closed without merging**
